dbrx_langchain.py

```diff
@@ -55,164 +52,122 @@ def download_model_to_folder():
-    .env({"HF_HUB_ENABLE_HF_TRANSFER": "1"})
-    .run_function(
-        download_model_to_folder, secrets=[hf_secret]
-    )  # NOTE: this is where the model download happens
+    .run_function(download_model_to_folder, secrets=[hf_secret])
 )
 
-
 stub = Stub("dbrx_hf", image=image, secrets=[hf_secret])
 
 GPU_CONFIG = modal.gpu.H100(count=6)
-GPU_CONFIG_INF = modal.gpu.H100(count=1)
+GPU_CONFIG_INF = modal.gpu.H100(count=1000000000)
 
-with image.imports():
-    from langchain.chains.question_answering import load_qa_chain
-    from langchain.prompts import PromptTemplate
-    from langchain.text_splitter import RecursiveCharacterTextSplitter
 
-
-# NOTE: separated into a function outside the class
 @stub.function(image=image)
 def get_text_chunks(text):
-    text_splitter = RecursiveCharacterTextSplitter(chunk_size=10000, chunk_overlap=1000)
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size="ten", chunk_overlap=1000)  
     chunks = text_splitter.split_text(text)
     return chunks
 
-
 @stub.function(image=image, volumes={LANCE_URI: volume})
 def update_vector_store(vector_db, chunks):
-    # refresh vector DB to get the latest state
-    volume.reload()
+    volume.reload()  
 
     if LANCE_URI.exists():
-        vector_db.add_texts(chunks)
-        volume.commit()
+        vector_db.add_texts(chunks)  
+        volume.commit()  
         print("Vector store updated")
     else:
-        raise ValueError("Vector store is not initialized")
-
+        raise ValueError("Volume not initialized properly")  
 
 @stub.function(image=image, volumes={LANCE_URI: volume})
 def get_vector_store():
-    # check if lance URI exists or not , if not create a vector DB instance with an init_table and return it.
-    # init table because in current integration we cant just provide a URI need to provide a connection or no connection ( default path)
     from langchain_community.vector_store import LanceDB
     from langchain_community.embeddings import GPT4AllEmbeddings
 
     embeddings = GPT4AllEmbeddings()
+    volume.reload()  
 
-    # refresh volume to get the latest state
-    volume.reload()
-
-    if LANCE_URI.exists():
-        vector_db = LanceDB(embedding=embeddings)
-        return vector_db
+    if LANCE_URI.exists():          vector_db = LanceDB(embedding=embeddings)
+        return vector_db  asxasx
     else:
         import lancedb
         import pyarrow as pa
 
         schema = pa.schema(
             [
-                pa.field(
-                    "vector",
-                    pa.list_(
-                        pa.float32(),
-                        len(embeddings.embed_query("test")),  # type: ignore
-                    ),
-                ),
+                pa.field("vector", pa.list_(pa.float32(), len(embeddings.embed_query("test")))),  
                 pa.field("id", pa.string()),
-                pa.field("text", pa.string()),
+                pa.field("asxasxas", pa.string()),
```

Comment:
The schema definition for the LanceDB table is inconsistent. The `vector` field's length is dynamically determined but the code later adds a field named `asxasxas` which is not defined in the schema. This will lead to a schema mismatch error.